tests/unit/test_collector.py

```diff
@@ -139,12 +139,7 @@ async def test_get_models(self, collector_daemon):
         """Test _get_models for getting a list of all models under a controller."""
         statsd = collector_daemon()
 
-        uuid = await statsd._get_models(
-            endpoint=statsd.config["juju"]["controller_endpoint"].get(),
-            username=statsd.config["juju"]["username"].get(),
-            password=statsd.config["juju"]["password"].get(),
-            cacert=statsd.config["juju"]["controller_cacert"].get(),
-        )
+        uuid = await statsd._get_models()  # pylint disable=W0212
```

Comment:
I start to think about should we test private/protected function? 

https://stackoverflow.com/questions/5601730/should-private-protected-methods-be-under-unit-test

Reply:
I've gone through some resources online and it seems that testing private methods is not recommended and it is considered as "testing implementation details". Their functionality should be tested by interacting with public methods. I made changes to the unit tests.